tests/MemoryLeakRepro/Program.fs

```diff
@@ -0,0 +1,82 @@
+﻿open System
+open System.Threading
+open System.Threading.Tasks
+open FSharp.Control
+open Propulsion.Feed
+open Propulsion.Internal
+open Propulsion.MessageDb
+open Propulsion.MessageDb.Internal
+open Propulsion.Sinks
+open Propulsion.Streams
+open Serilog
+
+Log.Logger <- LoggerConfiguration().WriteTo.Console().CreateLogger()
+
+let memoryCheckpoints =
+    { new IFeedCheckpointStore with
+        member _.Start(source, tranche, origin, ct) =
+            Task.FromResult((TimeSpan.FromSeconds 1000, Position.parse 0L))
+
+        member _.Commit(source, tranche, pos, ct) = Task.CompletedTask }
+
+// Running this results in a memory leak
+let rec propulsion () =
+
+    let sinkStats =
+        { new Propulsion.Streams.Stats<_>(Log.Logger, TimeSpan.FromMinutes 1, TimeSpan.FromMinutes 1) with
+            override _.HandleOk(_) = ()
+            override _.HandleExn(logger, x) = () }
+
+    let handle _ _ _ =
+        task { return struct (StreamResult.AllProcessed, ()) }
+
+    let sink = Factory.StartConcurrentAsync(Log.Logger, 2, 2, handle, sinkStats)
```

Comment:
Would be good if we can create a noop sink to verify that the problem isn't in the sink